features/step_definitions/user_login_steps.rb

```diff
@@ -10,6 +10,11 @@
   header 'X-GOODCITY-APP-NAME', "#{app}.goodcity"
 end
 
+Given(/^I am an old ([supervisor | reviewer | stock_fulfilment]+) logging into "([^"]*)" app/) do |role, app|
+  @user = create(:user, :"with_#{role}_role", :with_can_login_to_admin_permission, role_expiry: 5.days.ago)
+  header "X-GOODCITY-APP-NAME", "#{app}.goodcity"
+end
+
```

Comment:
This step can probably be merged with the one right above it by making "old" optional 

e.g `/^ I am an (old)? .../`

The 3 steps above are too similar to each others, ideally we want them as a single configurable step

If you want to push forward for now that's fine, but could you add a TODO comment to refactor later ? :)